service/history/transferQueueActiveProcessor.go

```diff
@@ -881,7 +897,12 @@ func (t *transferQueueActiveProcessorImpl) processRecordWorkflowStarted(task *pe
 	// release the context lock since we no longer need mutable state builder and
 	// the rest of logic is making RPC call, which takes time.
 	release(nil)
```

Comment:
here the lock is released.
on line 903 and line 906, at least one variable `searchAttr` is passed to another function.
since searchAttr is a map (pointer, not a copy) then there can be race case.

Reply:
Thanks for catching this, add deep copy